app/controllers/spree/checkout_controller.rb

```diff
@@ -47,6 +52,20 @@ def update
       end
     end
 
+    def registration
+      @user = Spree::User.new
+    end
+
+    def update_registration
+      if params[:order][:email] =~ Devise.email_regexp && current_order.update(email: params[:order][:email])
```

Comment:
Just double-checking, do we always have a `current_order` present?

Reply:
Forget it; I guess you just copied it here, so it should already have been checked.